### PR TITLE
Fix the problem with inner double quotes

### PR DESCRIPTION
**Pull request by OlegNadymov**
Do I correct create expression for my case with inner double quotes?
After move from System.Dynamic.Linq to System.Dynamic.Linq.Core all my similar expressions were broken

### CODE REVIEW

File: src/System.Linq.Dynamic.Core/Parser/ExpressionParser.cs [hunk 1]
```diff
@@ -755,6 +755,16 @@ Expression ParseStringLiteral()
             _textParser.ValidateToken(TokenId.StringLiteral);
             char quote = _textParser.CurrentToken.Text[0];
             string s = _textParser.CurrentToken.Text.Substring(1, _textParser.CurrentToken.Text.Length - 2);
+            int index1 = 0;
+            while (true)
+            {
+                int index2 = s.IndexOf(quote, index1);
+                if (index2 < 0)
```
Comment (StefH) [inline src/System.Linq.Dynamic.Core/Parser/ExpressionParser.cs]: Please use {} in this new code. Example:
```c#
if (index2 < 0)
{
    break;
}
```

File: src/System.Linq.Dynamic.Core/Parser/ExpressionParser.cs [hunk 2]
```diff
@@ -755,6 +755,16 @@ Expression ParseStringLiteral()
...
+                    break;
+                if (index2 + 1 < s.Length && s[index2 + 1] == quote)
+                    s = s.Remove(index2, 1);
```
Comment (StefH) [inline src/System.Linq.Dynamic.Core/Parser/ExpressionParser.cs]: Please use {} in this new code. Example:
```c#
if (index2 < 0)
{
    break;
}
```

File: src/System.Linq.Dynamic.Core/Parser/ExpressionParser.cs (hunk 1 above)
Comment (OlegNadymov) [reply src/System.Linq.Dynamic.Core/Parser/ExpressionParser.cs]: Ok! I'm done.

**Comment by OlegNadymov** [comment]
@StefH I made the required changes in the code. Something else needs to be done?

File: src/System.Linq.Dynamic.Core/Parser/ExpressionParser.cs (hunk 2 above)
Comment (OlegNadymov) [reply src/System.Linq.Dynamic.Core/Parser/ExpressionParser.cs]: Ok! It was done.

**Comment by StefH** [comment]
Thanks a lot.

A small question, how does the code behave if you use something like an incomplete escaped string as a search query? 
Example `test \"abc`.

**Comment by OlegNadymov** [comment]
@StefH may be you skipped my comment about new code.

**Comment by StefH** [comment]
Ok. I will merge the code soon.

**Comment by OlegNadymov** [comment]
@StefH ok! Thank you! And update nuget please.

**Comment by OlegNadymov** [comment]
@StefH did you update nuget?

**Comment by StefH** [comment]
Not yet sorry. Tomorrow morning.

**Comment by OlegNadymov** [comment]
@StefH thank you!!

**Pull request merged**